### DISCUSSION TITLE
padding with einops

### DISCUSSION THREAD

**Discussion by jakubMitura14** [arogozhnikov/einops#233] [Q&A] [answered]
Hello is it possible to perform 0 padding of a multidimensional tensor with einops ?

**Comment by arogozhnikov** [comment, marked as answer]
Hi @jakubMitura14,
einops does not contain tools for padding

**Reply by jakubMitura14** [reply]
thanks

**Comment by leshabirukov** [comment]
Well, if you really stubborn... Here's  padding, mode = ‘edge’

    top    = repeat( ims[:, 0,:,:], 'b w c -> b 32 w c' )
    bottom = repeat( ims[:,-1,:,:], 'b w c -> b 32 w c' )

    ims_pad_y, _ = pack( (top,ims,bottom), 'b * w c' )

    left   = repeat( ims_pad_y[:,:, 0,:], 'b h c -> b h 32 c' )
    right  = repeat( ims_pad_y[:,:,-1,:], 'b h c -> b h 32 c' )

    ims_pad, _ = pack( (left,ims_pad_y,right), 'b h * c' )

    rearrange(ims_pad, 'b h w c -> h (b w) c')   # just shows the result

Meaned to be run in context of 
https://nbviewer.org/github/arogozhnikov/einops/blob/master/docs/1-einops-basics.ipynb
